shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController.mm

```diff
@@ -0,0 +1,107 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/logging.h"
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController.h"
+
+@implementation FlutterPlatformViewController {
+  // NSDictionary maps platform view type identifiers to FlutterPlatformViewFactories.
+  NSMutableDictionary<NSString*, NSObject<FlutterPlatformViewFactory>*>* _platformViewFactories;
+
+  // Map from platform view id to the underlying NSView.
+  std::map<int, NSView*> _platformViews;
+
+  // View ids that are going to be disposed on the next present call.
+  std::unordered_set<int64_t> _platformViewsToDispose;
+}
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _platformViewFactories = [[NSMutableDictionary alloc] init];
+  }
+  return self;
+}
+
+- (void)onCreateWithViewID:(int64_t)viewId
+                  viewType:(nonnull NSString*)viewType
+                    result:(nonnull FlutterResult)result {
+  if (_platformViews.count(viewId) != 0) {
+    result([FlutterError errorWithCode:@"recreating_view"
+                               message:@"trying to create an already created view"
+                               details:[NSString stringWithFormat:@"view id: '%lld'", viewId]]);
+  }
+
+  NSObject<FlutterPlatformViewFactory>* factory = _platformViewFactories[viewType];
+  if (!factory) {
+    result([FlutterError
+        errorWithCode:@"unregistered_view_type"
+              message:@"trying to create a view with an unregistered type"
+              details:[NSString stringWithFormat:@"unregistered view type: '%@'", viewType]]);
+    return;
+  }
+
+  NSView* platform_view = [factory createWithFrame:CGRectZero viewIdentifier:viewId arguments:nil];
```

Comment:
If we always create with `createWithFrame:CGRectZero`, do we actually need this "frame" parameter at all?

Reply:
Tried to keep it inline with what iOS does: https://github.com/flutter/engine/blob/main/shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm#L158

Reply:
done!